lib/spack/spack/util/module_cmd.py

```diff
@@ -59,9 +59,14 @@ def module(*args):
                                      shell=True,
                                      executable="/bin/bash")
 
+        # Make sure the command succeeded
+        module_output = module_p.communicate()
+        if module_p.returncode != 0:
+            raise EnvironmentParseError(module_p.returncode, module_output[1])
```

Comment:
Looks like you want to return `module_output[0]` here for a meaningful error message, at least for the `command not found` error being reported here.

Reply:
Yep, the issue that led to #18624 only reported to stderr, but I'll go ahead and print both.